src/search_unused.rs

```diff
@@ -175,28 +159,18 @@ fn collect_paths(dir_path: &Path, analysis: &PackageAnalysis) -> Vec<PathBuf> {
     }
 
     // Collect all final paths for the crate first.
-    let paths: Vec<PathBuf> = root_paths
+    let paths = root_paths
         .iter()
         .flat_map(|root| WalkDir::new(dir_path.join(root)).into_iter())
         .filter_map(|result| {
-            let dir_entry = match result {
-                Ok(dir_entry) => dir_entry,
-                Err(err) => {
-                    eprintln!("{err}");
-                    return None;
-                }
-            };
-            if !dir_entry.file_type().is_file() {
-                return None;
-            }
-            if dir_entry
-                .path()
-                .extension()
-                .is_none_or(|ext| ext.to_string_lossy() != "rs")
-            {
-                return None;
-            }
-            Some(dir_entry.path().to_owned())
+            result
+                .inspect_err(|err| eprintln!("{err}"))
+                .ok()
+                .filter(|entry| {
+                    entry.file_type().is_file()
+                        && entry.path().extension().is_some_and(|ext| ext == "rs")
+                })
+                .map(|entry| entry.path().to_owned())
```

Comment:
Pro-tip: if you have a `.filter()` and then `.map()`, you can use `.filter_map()`. Let's do that here!

Reply:
I found it less readable, but if you insist.
You know you can change anything as a maintainer and refactor to our liking?